macros/src/main/scala/pureconfig/derivation/MacroCompat.scala

```diff
@@ -26,7 +26,10 @@ trait MacroCompat {
   // reported up to Scala 2.12.2. See https://github.com/scala/bug/issues/10398 for more information.
   def inferImplicitValueCompat(typ: Type): Tree = {
     val cc = c.asInstanceOf[scala.reflect.macros.contexts.Context]
-    val enclosingTree = cc.openImplicits.head.tree.asInstanceOf[cc.universe.analyzer.global.Tree]
+    val enclosingTree =
+      cc.openImplicits.headOption.map(_.tree)
+        .orElse(cc.enclosingMacros.lastOption.map(_.macroApplication))
```

Comment:
Hmm, I'm not familiar enough with the compiler internals anymore to predict the consequences of setting the enclosing tree as this expression... I hope everything works fine 😅

Reply:
Let's keep this then, the change makes sense to me and it doesn't seem very dangerous :)